sympy/matrices/common.py

```diff
@@ -772,9 +772,11 @@ def entry(i, j):
 
     @classmethod
     def _eval_eye(cls, rows, cols):
-        def entry(i, j):
-            return cls.one if i == j else cls.zero
-        return cls._new(rows, cols, entry)
+        from sympy.matrices import zeros
+        M = zeros(rows, cols)
+        for i in range(rows):
+            M[i*(1+rows)] = cls.one
+        return cls._new(M)
```

Comment:
This fails when `rows != cols`.

This could also be more efficient like:
```python
vals = [cls.zero]*(rows*cols)
vals[::cols+1] = [cls.one]*min(rows, cols)
M = cls._new(rows, cols, vals, copy=False)
```

Reply:
I am extremely sorry, I assumed it to be a `square matrix`.
Will use the code you provided.
Thanks.